### PR TITLE
SBOM UI feature implementation

### LINKED ISSUE #19132: [SBOM] UI implementation
The development work of the UI of SBOM in harbor.

### PR DESCRIPTION
**Pull request by xuelichao**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

Support to generate and down SBOM from harbor UI.

Mockups review link: (Access permission needed)
https://miro.com/app/board/uXjVN1ev40Q=/ 

# Issue being fixed
Fixes #19132

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/portal/src/app/base/project/repository/artifact/artifact.ts [hunk 1]
```diff
@@ -166,3 +166,148 @@
...
+    fieldPath: string,
+    data: Object,
+    value: Object
+) {
+    if (fieldPath && data) {
+        const keys = fieldPath?.split('.');
+        let tempData = data;
+        for (let i = 0; i < keys.length; i++) {
+            if (keys.length === i + 1) {
+                tempData[keys[i]] = value;
```
[the remark this answers is not in the document]
Comment (xuelichao) [reply src/portal/src/app/base/project/repository/artifact/artifact.ts]: fix

**Comment by Vad1mo** [comment]
@xuelichao can you attach the screenshots to the PR, not everyone in the community has access to the miro board.

**Comment by xuelichao** [comment]
Screenshot 2024-01-31 at 11 07 42
Screenshot 2024-01-26 at 12 06 16
Screenshot 2024-01-26 at 12 05 38

**Review by stonezdj — APPROVED** [review]

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**